code-challenges.test.js

```diff
@@ -22,33 +22,87 @@ var secretCodeWord1 = "lackadaisical"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
-
-
-
+describe("When encodeWord is called", () => {
+  it("return string with 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0. when a string is called",() => {
+    //arrange
+    const argument = secretCodeWord1
+    //act
+    const actualResult = encodeWord(argument)
+    //assert
+  expect(actualResult).toEqual("l4ck4d41s1c4l")
+  })
+  it("return string with 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0. when a string is called",() => {
+    //arrange
+    const argument = secretCodeWord2
+    //act
+    const actualResult = encodeWord(argument)
+    //assert
+  expect(actualResult).toEqual("g0bbl3dyg00k")
+  })
+})
 
 // b) Create the function that makes the test pass.
 
-
-
+//pseudocode
+//create a function that takes a string as an argument
+//.split the word into array
+//use .map to cycle through array of letters
+//use if else to check if contains letters and then changes them
+
+const encodeWord = (string) => {
+  letterArray = string.toLowerCase().split("")
+  return letterArray.map(letter => {
+    if(letter === "a"){
+    return 4
+    } else if(letter === "e"){
+    return 3
+    } else if(letter === "i"){
+    return 1
+    } else if(letter === "o"){
+    return 0
+    }else{
+    return letter
+    }
+  }).join("")
+}
 
 
 // --------------------2) Create a function that takes in an array and returns all the words that contain the letter a.
 
 // a) Create a test with an expect statement using the variable provided.
 
 var arrayOfWords = ["Apple", "Banana", "Plum", "Cherry", "Kiwi", "Peach"]
-// Expected output: "Apple" "Banana" "Peach"
-
+// Expected output: ["Apple" "Banana" "Peach"]
 
+describe("When aWordReturner is called", () => {
+  it("return array with all strings with letter a",() => {
+    //arrange
+    const argument = arrayOfWords
+    //act
+    const actualResult = aWordReturner(argument)
+    //assert
+  expect(actualResult).toEqual(["Apple", "Banana", "Peach"])
+  })
+})
 
 
 
 // b) Create the function that makes the test pass.
 
-
-
-
-
+//pseudocode
+//create function that takes in array as argument
+//.filter to cycle through elements in array and filter out words with letter a
+//.split each item in array
+//.includes to check if word contains the letter
+//return final array
+
+const aWordReturner = (array) => {
+  return array.filter(string => {
+    if(string.toLowerCase().split("").includes("a") === true){
+      return string
+    }
+  })
```

Comment:
best practice for using filter is to provide a function that returns a boolean but because the value returned on line 102 is evaluated to "truthy", it passes. A possible refactor could look like:

```suggestion
  return array.filter(string => string.toLowerCase().split("").includes("a"))
```